Prevent updateShadowCamera from corrupting shadowCam matrix.

riccardobl opened the pull request:
Prevent updateShadowCamera from corrupting shadowCam matrix when there isn't any visible shadow caster in scene. 
Discovered in #2111

stephengold commented:
Shouldn't the changes to asserts and diagnostics have been a separate PR?

Also, what's the significance of the hacktoberfest labels? (Just curious.)

riccardobl commented:
> Shouldn't the changes to asserts and diagnostics have been a separate PR?

Well i wouldn't dedicate a PR to this small change that i made just to debug this issue and I merged the pr without squash to keep the commits separated, so it shouldn't cause any issue for the git history or if we need to revert it.

> Also, what's the significance of the hacktoberfest labels? (Just curious.)

Two years ago, I assumed the responsibility of moderating [Hacktoberfest](https://hacktoberfest.com/) for our repositories. This year, in order for a pull request to be considered valid for the quota, it must either be approved, merged, or have a specific label applied for a continuous 7-day period. As we are in the second half of October, I've added the label to mark valid pull requests as soon as they are submitted, avoiding the need for rushed approvals in case the submitter is participating.
I added the label to this PR by mistake.

stephengold commented:
I see. Thanks for the answers.
